Shared/Tests.Shared/NotificationTests.cs

```diff
@@ -389,6 +389,100 @@ public void List_WhenTransactionHasBothAddAndRemove_ShouldReset()
+        [TestCase(0, 3, 4, 0, NotifyCollectionChangedAction.Reset)] // a b c d e -> e b c d a
+        [TestCase(0, 2, 0, 2, NotifyCollectionChangedAction.Move)] // a b c d e -> c a b d e
+        [TestCase(4, 2, 4, 2, NotifyCollectionChangedAction.Move)] // a b c d e -> a b d e c
+        [TestCase(1, 3, 1, 3, NotifyCollectionChangedAction.Move)] // a b c d e -> c d a b e
+        public void List_WhenTransactionHasMultipleMoves(int oldIndex1, int newIndex1, int oldIndex2, int newIndex2, NotifyCollectionChangedAction expectedAction)
+        {
+            OrderedObject object1 = null;
+            OrderedObject object2 = null;
+            var args = TestMoves(items =>
+            {
+                object1 = items[oldIndex1];
+                items.Move(object1, newIndex1);
+
+                object2 = items[oldIndex2];
+                items.Move(object2, newIndex2);
+            }, expectedAction);
+
+            if (expectedAction == NotifyCollectionChangedAction.Move)
+            {
+                var oldStartIndex = Math.Min(oldIndex1, oldIndex2);
+                var newStartIndex = Math.Min(newIndex1, newIndex2);
+                if (oldStartIndex < newStartIndex)
+                {
+                    // x was moved from before to after y, then y was moved to after x, which results in index being adjusted by -1.
+                    newStartIndex--;
+                }
+                else
+                {
+                    // x was moved from after to before y, then y was moved to before x, which results in index being adjusted by -1.
+                    oldStartIndex--;
+                }
+
+                Assert.That(args.OldStartingIndex, Is.EqualTo(oldStartIndex));
+                Assert.That(args.NewStartingIndex, Is.EqualTo(newStartIndex));
+                Assert.That(args.OldItems, Is.EquivalentTo(new[] { object1, object2 }));
+                Assert.That(args.NewItems, Is.EquivalentTo(new[] { object1, object2 }));
+            }
+        }
+
+        [TestCase(0, 4)]
+        [TestCase(4, 0)]
+        [TestCase(0, 2)]
+        [TestCase(2, 0)]
+        [TestCase(1, 2)]
+        [TestCase(2, 1)]
+        public void List_WhenTransactionHasSingleMove(int oldIndex, int newIndex)
+        {
+            OrderedObject movedObject = null;
+            var args = TestMoves(items =>
+            {
+                movedObject = items[oldIndex];
+                items.Move(movedObject, newIndex);
+            }, NotifyCollectionChangedAction.Move);
+
+            Assert.That(args.OldStartingIndex, Is.EqualTo(oldIndex));
+            Assert.That(args.NewStartingIndex, Is.EqualTo(newIndex));
+            Assert.That(args.OldItems, Is.EquivalentTo(new[] { movedObject }));
+            Assert.That(args.NewItems, Is.EquivalentTo(new[] { movedObject }));
+        }
+
+        private NotifyCollectionChangedEventArgs TestMoves(Action<IList<OrderedObject>> moveAction, NotifyCollectionChangedAction expectedAction)
```

Comment:
Please add a simple comment saying what you expect to be in the Realm as a result of this test helper.